src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudV1Spec.groovy

```diff
@@ -433,24 +432,20 @@ class FlowCrudV1Spec extends HealthCheckSpecification {
 
     def "A flow cannot be created with asymmetric forward and reverse paths"() {
         given: "Two active neighboring switches with two possible flow paths at least and different number of hops"
-        List<List<PathNode>> possibleFlowPaths = []
-        int pathNodeCount = 2
-        def (Switch srcSwitch, Switch dstSwitch) = topology.getIslsForActiveSwitches().find {
-            possibleFlowPaths = database.getPaths(it.srcSwitch.dpId, it.dstSwitch.dpId)*.path.sort { it.size() }
-            possibleFlowPaths.size() > 1 && possibleFlowPaths.max { it.size() }.size() > pathNodeCount
-        }.collect {
-            [it.srcSwitch, it.dstSwitch]
-        }.flatten() ?: assumeTrue(false, "No suiting active neighboring switches with two possible flow paths at least and " +
+        List<List<Isl>> availablePathsIsls
+        int shortestIslCountPath = 1
+        def swPair = switchPairs.all().neighbouring().withAtLeastNNonOverlappingPaths(2).getSwitchPairs().find {
+            availablePathsIsls = it.paths.collect { it.getInvolvedIsls() }
+            availablePathsIsls.size() > 1 && availablePathsIsls.find { it.size() > shortestIslCountPath}
```

Comment:
There is no need to find Path with max nodes count as 
`possibleFlowPaths.max { it.size() }.size() > pathNodeCount`  can be simplify to find any Path with more that expected nodes count (isl number) `availablePathsIsls.find { it.size() > shortestIslCountPath}`